### ISSUE TITLE
ncdataset/readdata errors with .grb2 file

### ISSUE THREAD

**Issue by sarahla1** [nctoolbox/nctoolbox#91]
Hello,
New to working with nctoolbox.  Read through other issues, but haven't found a solution yet.

_____________________________________________________________________
Here's pertinent section of my code: 

data = ncgeodataset(FileName);
PRE(n,:)  = (data{'isobaric'}(:))';
var = data{'u-component_of_wind_isobaric'}(:);
WND(n).U = squeeze(var(1,:,latIdx,lonIdx));
var = data{'v-component_of_wind_isobaric'}(:);
WND(n).V = squeeze(var(1,:,latIdx,lonIdx));
_____________________________________________________________________
Here's the error message:

Error using ncdataset/readdata (line 491)
Java exception occurred:
ucar.ma2.InvalidRangeException: last (0) must be >= first (57)

	at ucar.ma2.Range.<init>(Range.java:146)

	at ucar.ma2.Range.<init>(Range.java:130)

Error in ncdataset/data (line 171)
                    d = obj.readdata(variable, first, last, stride);

Error in ncgeodataset/subsref (line 630)
                            B = builtin('subsref',obj,g);

Error in ncvariable/somedata (line 406)
                data = obj.dataset.data(name, first, last, stride);

Error in ncvariable/subsref (line 292)
                                        sref = obj.somedata(1, first, last, stride);

Error in ncgeodataset/subsref (line 660)
                        B = v.data(s(2).subs{:});

Error in ReadWeather (line 36)
var = data{'v-component_of_wind_isobaric'}(:);
______________________________________________________________
>> data.variables

ans = 

    'u-component_of_wind_isobaric'
    'v-component_of_wind_isobaric'
    'lat'
    'lon'
    'isobaric'
.
.
.
______________________________________________________________
Here is my configuration:

>> ver
-------------------------------------------------------------------------------------------------------
MATLAB Version: 8.0.0.783 (R2012b)
MATLAB License Number: 1082431
Operating System: Microsoft Windows 7 Version 6.1 (Build 7601: Service Pack 1)
Java Version: Java 1.6.0_17-b04 with Sun Microsystems Inc. Java HotSpot(TM) 64-Bit Server VM mixed mode
-------------------------------------------------------------------------------------------------------
MATLAB                                                Version 8.0        (R2012b)
NCTOOLBOX Tools for read-only access to Common ...    Version 1.1.1-13-g8582810++
______________________________________________________________

The same code and .grb2 file works fine on Linux & Windows Server 2008 Operating Systems.

If I type in the error producing line,
   var = data{'v-component_of_wind_isobaric'}(:);
in matlab command window, I do NOT get the error message.

It's when I run the .m file, I get the error message.
______________________________________________________________
In case this provides any clue, here are my java paths:

>> javaclasspath

		STATIC JAVA PATH

	C:\Program Files\MATLAB\R2012b\java\patch                                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\AnimatedTransitions.jar                              
	C:\Program Files\MATLAB\R2012b\java\jarext\ant.jar                                              
	C:\Program Files\MATLAB\R2012b\java\jarext\ant-launcher.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\antlr.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\felix.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\scr.jar                                              
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\activation.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\axiom-api.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\axiom-impl.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\axis2-adb.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\axis2-kernel.jar                               
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\axis2-transport-http.jar                       
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\axis2-transport-local.jar                      
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\commons-httpclient.jar                         
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\geronimo-stax-api.jar                          
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\httpcore.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\mail.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\neethi.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\wsdl4j.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\wstx-asl.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\axis2\XmlSchema.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-anim.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-extension.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-dom.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-transcoder.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-awt-util.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-bridge.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-gvt.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-css.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-parser.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-ext.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-gui-util.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-svggen.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-codec.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-svg-dom.jar                                    
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-script.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-util.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-swing.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\batik-xml.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\bluecove-bluez.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\bluecove-emu.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\bluecove.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-codec.jar                                    
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-collections.jar                              
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-compress.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-io.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-lang.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-logging.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-math.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\commons-net.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\dom4j.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\fop.jar                                              
	C:\Program Files\MATLAB\R2012b\java\jarext\avalon-framework.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\foxtrot.jar                                          
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-io.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-util.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-xml.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphics2d.jar                               
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphicsio.jar                               
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphicsio-pdf.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphicsio-emf.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphicsio-ps.jar                            
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphicsio-swf.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphicsio-svg.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\freehep-graphics-base.jar                            
	C:\Program Files\MATLAB\R2012b\java\jarext\glazedlists_java15.jar                               
	C:\Program Files\MATLAB\R2012b\java\jarext\google-collect.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\guice\aopalliance.jar                                
	C:\Program Files\MATLAB\R2012b\java\jarext\guice\guice.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\guice\guice-assistedinject.jar                       
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ice.jar                                          
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6core.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6extra.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6http.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6https.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6js.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6swing.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6util.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\ib6xalan.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\ice\icessl.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\access-bridge.jar                                    
	C:\Program Files\MATLAB\R2012b\java\jarext\jaccess-1_4.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\jaxen.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\jdom.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jarext\annotations.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\jgoodies-forms.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\jgoodies-looks.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-action.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-common.jar                                 
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-components.jar                             
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-dialogs.jar                                
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-dock.jar                                   
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-properties.jar                             
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-grids.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\jide\jide-shortcut.jar                               
	C:\Program Files\MATLAB\R2012b\java\jarext\jogl.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jarext\gluegen-rt.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\engine-gecko.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\engine-ie.jar                              
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\engine-webkit.jar                          
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\jxbrowser.jar                              
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\MozillaInterfaces.jar                      
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\runtime.jar                                
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\slf4j-api.jar                              
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\slf4j-log4j12.jar                          
	C:\Program Files\MATLAB\R2012b\java\jarext\jxbrowser\winpack.jar                                
	C:\Program Files\MATLAB\R2012b\java\jarext\jxlayer.jar                                          
	C:\Program Files\MATLAB\R2012b\java\jarext\log4j.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\lucene-analyzers-2.0.0.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\lucene-core-2.0.0.jar                                
	C:\Program Files\MATLAB\R2012b\java\jarext\nekohtml.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jarext\openxml4j.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jarext\RXTXcomm.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-ant.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-dom.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-dom4j.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-jdom.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-s9api.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-sql.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-xom.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-xpath.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9-xqj.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon9.jar                                           
	C:\Program Files\MATLAB\R2012b\java\jarext\tablelayout.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\TimingFramework.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jarext\webrenderer.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jarext\xercesImpl.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\xml-apis-ext.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\xmlgraphics-commons.jar                              
	C:\Program Files\MATLAB\R2012b\java\jarext\xstream.jar                                          
	C:\Program Files\MATLAB\R2012b\java\jar\instwiz.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\appmanagement.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\coder.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jar\comparisons.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\fixedpoint.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jar\foundation_libraries.jar                                
	C:\Program Files\MATLAB\R2012b\java\jar\install.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\instrument.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jar\lmlogin.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\mde.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\matlab.jar                                              
	C:\Program Files\MATLAB\R2012b\java\jar\jmi.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\services.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jar\widgets.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\mlservices.jar                                          
	C:\Program Files\MATLAB\R2012b\java\jar\timer.jar                                               
	C:\Program Files\MATLAB\R2012b\java\jar\fatalexit.jar                                           
	C:\Program Files\MATLAB\R2012b\java\jar\mvm.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\project\api.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jar\project\impl.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jar\mlwidgets.jar                                           
	C:\Program Files\MATLAB\R2012b\java\jar\hg.jar                                                  
	C:\Program Files\MATLAB\R2012b\java\jar\ide.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\matlab\guide.jar                                
	C:\Program Files\MATLAB\R2012b\java\jar\page.jar                                                
	C:\Program Files\MATLAB\R2012b\java\jarext\webservices\mwlic_client\mw-license-server-client.jar
	C:\Program Files\MATLAB\R2012b\java\jar\util.jar                                                
	C:\Program Files\MATLAB\R2012b\java\jar\common.jar                                              
	C:\Program Files\MATLAB\R2012b\java\jar\mwswing.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\beans.jar                                               
	C:\Program Files\MATLAB\R2012b\java\jar\mwt.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\xml.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\desktop.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\toolstrip.jar                                           
	C:\Program Files\MATLAB\R2012b\java\jar\org\netbeans.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jar\org\openide.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jarext\saxon.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\helpsearch.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jar\help\helpui.jar                                         
	C:\Program Files\MATLAB\R2012b\java\jar\html.jar                                                
	C:\Program Files\MATLAB\R2012b\java\jarext\search.jar                                           
	C:\Program Files\MATLAB\R2012b\java\jar\product.jar                                             
	C:\Program Files\MATLAB\R2012b\java\jar\activationclient.jar                                    
	C:\Program Files\MATLAB\R2012b\java\jar\deactivation.jar                                        
	C:\Program Files\MATLAB\R2012b\java\jar\vrd.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\publishparser.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\shared\controllib.jar                           
	C:\Program Files\MATLAB\R2012b\java\jar\instutil.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\rptgencore.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\testmeas.jar                                    
	C:\Program Files\MATLAB\R2012b\java\jar\activation.jar                                          
	C:\Program Files\MATLAB\R2012b\java\jar\toolbox\timeseries.jar                                  
	C:\Program Files\MATLAB\R2012b\java\jar\webintegration.jar                                      
	C:\Program Files\MATLAB\R2012b\java\jar\mlwebservices.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jar\net.jar                                                 
	C:\Program Files\MATLAB\R2012b\java\jar\webproxy.jar                                            
	C:\Program Files\MATLAB\R2012b\java\jarext\mwaws_client.jar                                     
	C:\Program Files\MATLAB\R2012b\java\jarext\dws_client.jar                                       
	C:\Program Files\MATLAB\R2012b\java\jarext\webservices\loginws_client.jar                       
	C:\Program Files\MATLAB\R2012b\java\jarext\webservices\service_request_client.jar               
	C:\Program Files\MATLAB\R2012b\java\jarext\webservices\dws_client.jar                           
	C:\Program Files\MATLAB\R2012b\java\jarext\webservices\ws_client_core\mw-service-client-core.jar

		DYNAMIC JAVA PATH

	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\udunits-4.3.15.jar      
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\slf4j-log4j12-1.7.2.jar 
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\quartz-2.1.6.jar        
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\protobuf-java-2.4.1.jar 
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\opendap-4.3.15.jar      
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\netcdf-4.3.15.jar       
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\log4j-1.2.17.jar        
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\joda-time-2.1.jar       
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\jna-3.0.9.jar           
c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\je-4.0.92.jar           
c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\jcl-over-slf4j-1.7.2.jar
c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\jcip-annotations-1.0.jar
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\grib-4.3.15.jar         
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\ehcache-core-2.6.5.jar  
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java\bufr-4.3.15.jar         
	c:\users\s111\desktop\weather\nctoolbox-1.1.3\java    
______________________________________________________________

I'd really appreciate any help you can provide.
Thank you.
sarah

**Comment by rsignell-usgs** [comment]
So you are using exactly the same version of nctoolbox on both systems?

**Comment by sarahla1** [comment]
yes, nctoolbox version 1.1.3 on the "error-free" operating systems (Linux & Windows Server 2008) and "error-producing" Windows 7.  On Windows 7, I also tried other versions of nctoolbox (1.1.0 through 1.1.2), and got same errors.

**Comment by watertcafe** [comment]
Same here. Does anyone find a solution? I am using Matlab 2016b on a Windows 7 machine

**Comment by sbruceYoung** [comment]
what is more weird is that I can read one grib file success but occur the error while reading another grib file

**Comment by rsignell-usgs** [comment]
Can you try opening this file with the latest version of netcdf-java ToolsUI and report back?
https://www.youtube.com/watch?v=GTFvdhlHivM

**Comment by sbruceYoung** [comment]
Thank you for your attention, rsignell-usgs. hohonuuli had tried with the latest version of the NetCDF-Java toolsUI-4.6.10.jar. toolsUI throws an ArrayIndexOutOfBoundsException
ava.lang.ArrayIndexOutOfBoundsException: 315258
	at ucar.nc2.grib.QuasiRegular.linear(QuasiRegular.java:276)
	at ucar.nc2.grib.QuasiRegular.convertQuasiGrid(QuasiRegular.java:151)
	at ucar.nc2.grib.grib1.Grib1Record.readData(Grib1Record.java:183)
	at ucar.nc2.grib.grib1.Grib1Record.readData(Grib1Record.java:167)
	at ucar.nc2.grib.grib1.Grib1Record.readData(Grib1Record.java:230)
	at ucar.nc2.grib.collection.Grib1Iosp.readData(Grib1Iosp.java:447)
	at ucar.nc2.grib.collection.GribIosp$DataReader.read(GribIosp.java:1096)
	at ucar.nc2.grib.collection.GribIosp.readDataFromCollection(GribIosp.java:1018)
	at ucar.nc2.grib.collection.GribIosp.readData(GribIosp.java:963)
	at ucar.nc2.NetcdfFile.readData(NetcdfFile.java:2020)
	at ucar.nc2.Variable.reallyRead(Variable.java:874)
	at ucar.nc2.Variable._read(Variable.java:845)
	at ucar.nc2.Variable.read(Variable.java:723)
	at ucar.nc2.ui.VariablePlot.setVariable(VariablePlot.java:269)
	at ucar.nc2.ui.DatasetViewer.dataPlot(DatasetViewer.java:757)
	at ucar.nc2.ui.DatasetViewer.access$1400(DatasetViewer.java:78)
	at ucar.nc2.ui.DatasetViewer$NestedTable$6.actionPerformed(DatasetViewer.java:456)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
       ...

**Comment by Toby-D** [comment]
tl;dr: For certain variables, use the syntax `lon=nc{'lon'}(1);` instead of `lon=nc{'lon'}(:);` to prevent this error.

-------------------------------------------
I encountered the same error as the OP when I tried to convert 2D nc data to an array of singles using the syntax
It seems like the ':' input gets read in as the numeric value of the ':' character (which becomes 58) rather than as a special character as in MATLAB. Not sure if this is dependent on Java versions.

I got the same error as the OP when using this syntax:
```
nc=ncgeodataset(url);
nv=nc{'nv'}(:).';
```
where nv is a 2D dataset. However, I also found that while the colon syntax doesn't work as expected even for a 1D dataset:
`lon=nc{'lon'}(1);`
cuts off the first 57 datapoints in the series, because 'obj.readdata' converts ':' to the single 58. 

The solution is to specify the input argument as [1 1], and let 'obj.readdata' automatically find the max dimensions of the variable, e.g. 

```
lat=nc{'lat'}(1);
nv=nc{'nv'}([1 1])'
```
for 1D and 2D datasets, respectively.
